src/Http/Routing/src/Matching/DfaMatcherBuilder.cs

```diff
@@ -111,6 +113,46 @@ public DfaNode BuildDfaTree(bool includeLabel = false)
         return root;
     }
 
+    private sealed class CachingParameterPolicyFactory : ParameterPolicyFactory
+    {
+        private readonly ParameterPolicyFactory _inner;
+        private readonly Dictionary<string, IParameterPolicy> _cachedParameters;
+
+        public CachingParameterPolicyFactory(ParameterPolicyFactory inner)
+        {
+            _inner = inner;
+            _cachedParameters = new Dictionary<string, IParameterPolicy>();
```

Comment:
should we specify the text comparer here rather than using the default? presumably `StringComparer.Ordinal` or `StringComparer.InvariantCulture`?